x-pack/plugins/maps/public/classes/fields/table_field.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FIELD_ORIGIN } from '../../../common/constants';
+import { IField, AbstractField } from './field';
+import { IVectorSource } from '../sources/vector_source';
+import { TableSource } from '../sources/table_source';
+
+export class TableField extends AbstractField implements IField {
```

Comment:
This class is identical to `GeoJsonFileField` with the only difference being type of `_source`. Why not use generics and then create a single field class? Maybe call it something like `InlineField`?